tlslite/keyexchange.py

```diff
@@ -292,6 +297,38 @@ def makeClientKeyExchange(self):
         cke.createDH(self.dh_Yc)
         return cke
 
+# the DHE_RSA part comes from IETF ciphersuite names, we want to keep it
+#pylint: disable = invalid-name
+class DHE_RSAKeyExchange(ADHKeyExchange):
+    """
+    Handling of ephemeral Diffe-Hellman Key exchange
+
+    NOT stable API, do NOT use
+    """
+
+    def __init__(self, cipherSuite, clientHello, serverHello, privateKey):
+        super(DHE_RSAKeyExchange, self).__init__(cipherSuite, clientHello,
+                                                 serverHello, privateKey)
+#pylint: enable = invalid-name
+        self.dh_Xs = None
+        self.dh_Yc = None
+
+    # RFC 3526, Section 8.
+    strength = 160
+
+    def makeServerKeyExchange(self, sigHash=None):
+        """Prepare server side of key exchange with selected parameters"""
+        # Per RFC 3526, Section 1, the exponent should have double the entropy
+        # of the strength of the curve.
+        self.dh_Xs = bytesToNumber(getRandomBytes(self.strength * 2 // 8))
+        dh_Ys = powMod(self.dh_g, self.dh_Xs, self.dh_p)
+
+        version = self.serverHello.server_version
+        serverKeyExchange = ServerKeyExchange(self.cipherSuite, version)
+        serverKeyExchange.createDH(self.dh_p, self.dh_g, dh_Ys)
```

Comment:
can't all of the above in this method be replaced with a `super(DHE_RSAKeyExchange, self).makeServerKeyExchange()`?